src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session_ComputeModel.cs

```diff
@@ -103,119 +101,44 @@ public void Do()
 
                         // TODO(cyrusn): We're calling into extensions, we need to make ourselves resilient
                         // to the extension crashing.
-                        var groups = await GetGroupsAsync(_completionService, _triggerInfo, cancellationToken).ConfigureAwait(false);
-                        if (groups == null)
+                        var completionList = await GetCompletionListAsync(_completionService, _triggerInfo, cancellationToken).ConfigureAwait(false);
+                        if (completionList == null)
                         {
                             return null;
                         }
 
-                        groups.Do(AddGroupToMap);
-                        if (_displayNameToItemsMap.Count == 0)
-                        {
-                            return null;
-                        }
-
-                        var totalItems = _displayNameToItemsMap.Values.Flatten().ToList();
-                        totalItems.Sort();
-
                         var trackingSpan = await _completionService.GetDefaultTrackingSpanAsync(_documentOpt, _subjectBufferCaretPosition, cancellationToken).ConfigureAwait(false);
                         return Model.CreateModel(
                             _disconnectedBufferGraph,
                             trackingSpan,
-                            totalItems,
-                            selectedItem: totalItems.First(),
+                            completionList.Items.ToList(),
```

Comment:
Yeah, I don't like this either. I'll be making Items an ImmutableArray in a future change. That's the plan.